.github/workflows/build-deploy-dev.yml

```diff
@@ -5,42 +5,133 @@ on:
+    if: github.event.pull_request.merged == true || github.event_name == 'workflow_dispatch'
     steps:
       - name: Print Env Vars
         run: |
           echo Git Base Ref: ${{ github.base_ref }}
-          echo Git Build ID: ${{ github.event.number }}
           echo Git Pull Request Ref: ${{ github.event.pull_request.head.sha }}
           echo OC CLI Version: $(oc version)
 
+  # Create new tag.
+  createTag:
+    name: "Create tag"
+    if:  github.event.pull_request.merged == true || github.event_name == 'workflow_dispatch' 
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get bumb value
+        uses: haya14busa/action-cond@v1
+        id: getBumb
+        with:
+          cond: ${{ github.event_name == 'workflow_dispatch'}}
+          if_true: ${{ inputs.bumb }}
+          if_false: 'prerelease'
+
+      - name: Get bumb value, if provided
+        uses: haya14busa/action-cond@v1
+        id: getPreId
+        with:
+          cond: ${{ github.event_name == 'workflow_dispatch'}}
+          if_true: ${{ inputs.preId }}
+          if_false: 'prerelease'
+      - run: |
+          echo "bumb: ${{ steps.getBumb.outputs.value }}"
+          echo "prefix: ${{ steps.getPreId.outputs.value }}"
+  
+      - id: createNextTag
+        name: "Create tag"
+        uses: zwaldowski/semver-release-action@v3
+        with: 
+          prefix: "v"
+          preid: ${{ steps.getPreId.outputs.value }}       
+          bump:  ${{ steps.getBumb.outputs.value }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+      - run: echo "Tag ${{ steps.createNextTag.outputs.version_tag }} created!"
+    outputs:
+      newTag: ${{ steps.createNextTag.outputs.version_tag }}
+
+
+  # Create new release.
+  createRelease:
+    name: "Create release"
+    if: github.event_name == 'workflow_dispatch' && (inputs.releaseType == 'publish' || inputs.releaseType == 'draft')
+    runs-on: ubuntu-latest
+    needs: createTag
+    steps:
+      - name: Create release
+        id: createNewRelease
+        uses: Roang-zero1/github-create-release-action@v3
+        with:
+          created_tag: ${{ needs.createTag.outputs.newTag }}
+          create_draft: ${{ inputs.releaseType == 'draft' }}
+          release_title:  Release ${{ needs.createTag.outputs.newTag }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: echo "created release ${{ needs.createTag.outputs.newTag }} ${{ steps.createNewRelease.outputs.html_url }}!"
+
+
   # Building all packages
   build:
     name: Building all packages
     environment: DEV
     runs-on: ubuntu-latest
-    if: ${{ github.event.pull_request.merged == true}}
+    needs: createTag
+    if: github.event.pull_request.merged == true || github.event_name == 'workflow_dispatch' 
     env:
-      BUILD_ID: ${{ github.event.number }}
       BUILD_NAMESPACE: 0c27fb-tools
-      BUILD_REF: ${{ github.base_ref }}
+      BUILD_REF: ${{ needs.createTag.outputs.newTag }}
```

Comment:
does ${{ createTag.outputs.newTag }} does not resolve the build_ref? do we need "needs"?

Reply:
we are building the new tag created in the previous job `createTag`. I am not sure, if i got your qn

Reply:
as we did the test, `need` is needed